### ISSUE TITLE
datetime datapoints to pile up with webgl enabled

### ISSUE THREAD

**Issue by anadyn** [bokeh/bokeh#3795]
In Bokeh 0.11 there is some kind of truncation error in the rendering of a datetime axis when webgl=True. It causes the datapoints to pile up at discrete times.

Consider this minimal example:

```python
import numpy as np
import bokeh.plotting as bk
from datetime import datetime,timedelta

no_of_datapoints=10

base = datetime(2016, 2, 1, 9, 36, 0)
date_list = [base - timedelta(minutes=x) for x in range(0, no_of_datapoints)]
datapoints=np.arange(no_of_datapoints)

p=bk.figure(webgl=True,x_axis_type="datetime")
p.scatter(x=date_list,y=datapoints)
bk.show(p)
```

With webgl=True the diagram gets piled up datapoints:
![qvm85ci](https://cloud.githubusercontent.com/assets/1796208/12834483/9468b64c-cb5c-11e5-8ca3-e266d90501a2.png)

With webgl=False the diagram is rendered correctly 
![wxt6mwa](https://cloud.githubusercontent.com/assets/1796208/12834492/a577902a-cb5c-11e5-8778-04745879f9d5.png)

See also this [stackoverflow discussion](http://stackoverflow.com/questions/35113105/visualizing-timeseries-with-bokeh-datapoints-overlap/35126714)

**Comment by birdsarah** [comment]
Thanks for the report @anadyn, looks like a bug to me.

ping @almarklein

**Comment by birdsarah** [comment]
confirmed on 0.11.1: http://nbviewer.jupyter.org/github/birdsarah/bokeh-miscellany/blob/master/tests/3795%20webgl%20pile-up.ipynb

**Comment by anadyn** [comment]
I am glad to be able to contribute! Thanks for tidying up my post and clarifying the title @birdsarah.

**Comment by almarklein** [comment]
This sounds familiar. I managed to track down an earlier similar case (that we fixed): #2956. Will have a look soon.

**Comment by almarklein** [comment]
In this case, the problem is caused by float32 precision problems. Got it fixed though.
